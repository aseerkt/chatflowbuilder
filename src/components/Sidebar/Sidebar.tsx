import useStore from '@/stores'
import { nodeSelector } from '@/stores/selectors'
import { NodePanel } from '../NodePanel'
import { SettingsPanel } from '../SettingsPanel'

export default function Sidebar() {
  const { nodeId, nodeType } = useStore(nodeSelector)

  return (
    <aside className='w-[350px] border-l'>
      {nodeId && nodeType ? (
        <SettingsPanel nodeId={nodeId} nodeType={nodeType} />
      ) : (
        <NodePanel />
      )}
    </aside>
  )
}
