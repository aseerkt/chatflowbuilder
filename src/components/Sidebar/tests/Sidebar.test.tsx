import useStore from '@/stores'
import { render } from '@testing-library/react'
import Sidebar from '../Sidebar'

describe('<Sidebar /> tests', () => {
  it('should render <NodePanel />', () => {
    const { baseElement } = render(<Sidebar />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should render <SettingsPanel />', () => {
    useStore.setState({
      nodeId: 'node_1',
      nodeType: 'textNode',
      nodes: [
        {
          id: 'node_1',
          data: {
            message: 'message 1',
          },
          position: {
            x: 150,
            y: 100,
          },
        },
      ],
    })
    const { baseElement } = render(<Sidebar />)
    expect(baseElement).toMatchSnapshot()
  })
})
