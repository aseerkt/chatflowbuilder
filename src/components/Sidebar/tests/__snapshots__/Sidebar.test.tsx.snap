// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Sidebar /> tests > should render <NodePanel /> 1`] = `
<body>
  <div>
    <aside
      class="w-[350px] border-l"
    >
      <div
        class="p-3"
      >
        <div
          class="flex w-[200px] cursor-grab flex-col items-center gap-y-3 rounded-md border-2 border-green-600 p-3 text-green-600 shadow-sm"
          draggable="true"
        >
          <svg
            class="tabler-icon tabler-icon-message "
            fill="none"
            height="56"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="56"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 9h8"
            />
            <path
              d="M8 13h6"
            />
            <path
              d="M18 4a3 3 0 0 1 3 3v8a3 3 0 0 1 -3 3h-5l-5 3v-3h-2a3 3 0 0 1 -3 -3v-8a3 3 0 0 1 3 -3h12z"
            />
          </svg>
          <span>
            Message
          </span>
        </div>
      </div>
    </aside>
  </div>
</body>
`;

exports[`<Sidebar /> tests > should render <SettingsPanel /> 1`] = `
<body>
  <div>
    <aside
      class="w-[350px] border-l"
    >
      <div>
        <header
          class="flex items-center gap-x-4 border-b px-4 py-2"
        >
          <button
            aria-label="Back to node panel"
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 rounded-full p-0"
          >
            <svg
              class="tabler-icon tabler-icon-arrow-left "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12l14 0"
              />
              <path
                d="M5 12l6 6"
              />
              <path
                d="M5 12l6 -6"
              />
            </svg>
          </button>
          <span
            class="font-semibold"
          >
            Message
          </span>
        </header>
        <div
          class="flex flex-col gap-y-2 p-4"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for="message"
          >
            Text
          </label>
          <textarea
            class="flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id="message"
            rows="4"
          >
            message 1
          </textarea>
        </div>
      </div>
    </aside>
  </div>
</body>
`;
